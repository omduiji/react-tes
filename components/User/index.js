import React from 'react'
import { getDisplayName } from '../../utils'
import styles from './styles.module.scss'

const User = ({ user, setUser, active }) => (
    <li className={[styles.user, active && styles.active].filter(c => c).join(' ')}>
        <button type="button" onClick={e => setUser(user)}>
            <img src={user.photo} />
            <span>{getDisplayName(user)}</span>
        </button>
    </li>
)

export default User
