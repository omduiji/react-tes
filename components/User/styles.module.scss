.user {
    margin: 0;
    width: 100%;

    button {
        width: 100%;
        margin: 0;
        background-color: transparent;
        color: inherit;
    }

    img {
        max-width: 50px;
        margin-right: 10px;
    }

    &.active button {
        background-color: #f8f8f8;
    }

    &:not(:last-child) {
        border-bottom: 1px solid #eee;
    }
}
