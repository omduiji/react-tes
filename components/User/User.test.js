import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import User from '.'

const setUser = jest.fn()

const testUser = {
    id: 42,
    photo: 'test.jpg',
    first_name: 'Jane',
    last_name: 'Doe',
}

test('should render User', () => {
  const { queryByText, asFragment } = render(<User user={testUser} setUser={setUser} />)

  expect(queryByText('JaneDoe')).toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot()
})

test('should render active User', () => {
    const { queryByText, asFragment } = render(<User user={testUser} setUser={setUser} active />)

    expect(queryByText('JaneDoe')).toBeInTheDocument()

    expect(asFragment()).toMatchSnapshot()
})

test('should call setUser', () => {
    const { queryByText } = render(<User user={testUser} setUser={setUser} active />)

    fireEvent.click(queryByText('JaneDoe'))

    expect(setUser).toHaveBeenCalled()
})
