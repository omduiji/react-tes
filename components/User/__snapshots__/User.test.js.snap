// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render User 1`] = `
<DocumentFragment>
  <li
    class="user"
  >
    <button
      type="button"
    >
      <img
        src="test.jpg"
      />
      <span>
        JaneDoe
      </span>
    </button>
  </li>
</DocumentFragment>
`;

exports[`should render active User 1`] = `
<DocumentFragment>
  <li
    class="user active"
  >
    <button
      type="button"
    >
      <img
        src="test.jpg"
      />
      <span>
        JaneDoe
      </span>
    </button>
  </li>
</DocumentFragment>
`;
